src/app/shell/personal-cabinet/applications/application-card/application-card.component.ts

```diff
@@ -118,6 +138,41 @@ export class ApplicationCardComponent implements OnInit {
     });
   }
 
+    /**
+   * This method emit reject Application
+   * @param Application application
+   */
+  onBlock(blockedParent: BlockedParent): void {
+    const dialogRef = this.matDialog.open(BlockModalWindowComponent, {});
+    dialogRef.afterClosed().subscribe((result: string)  => {
+      if(result) {
+        const provider: Provider = this.store.selectSnapshot<Provider>(RegistrationState.provider);
+        const parent = this.application.parentId;
+        const reason = result;
+        blockedParent = new BlockedParent( parent, provider.id, reason);
```

Comment:
let's pass it directly to constructor, so we can remove additional value reason 
  blockedParent = new BlockedParent( parent, provider.id, result);